src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -633,21 +634,61 @@ def role_list(_):
         roles = rgw_client.list_roles()
         return roles
 
+    @staticmethod
+    def role_create(_, role_name: str = '', role_path: str = '', role_assume_policy_doc: str = ''):
+        assert role_name
+        assert role_path
+        rgw_client = RgwClient.admin_instance()
+        rgw_client.create_role(role_name, role_path, role_assume_policy_doc)
+        return f'Role {role_name} created successfully'
+
+
+# pylint: disable=C0301
+assume_role_policy_help = (
+    'Paste a json assume role policy document, to find more information on how to get this document, <a '  # noqa: E501
+    'href="https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-assumerolepolicydocument"'  # noqa: E501
+    '>click here.</a>'
+)
```

Comment:
As we start to display text from the back-end directly into the front-end, and our front-end (Angular) is committed to support i18n, then we should [start tagging these strings for i18n](https://docs.python.org/3/library/i18n.html), merge those Gettext .mo files with the Angular ones (we're using XLIFF format in Angular, but [Angular supports others](https://angular.io/guide/i18n-common-translation-files#change-the-source-language-file-format)) or perform separate translation, and then create the corresponding translated Gettext .po files to the back-end side. We could explore alternative Python i18n if it supports XLIFF out of the box

Reply:
I saw this [i18n](https://formly.dev/docs/examples/advanced/i18n) option in the formly which internally uses the ngx-translate to do the internationalization. Its there for the labels (props), so maybe this can be used for our internationalization purposes as well.

Reply:
There's an [alternative approach](https://formly.dev/docs/examples/advanced/i18n-alternative) as well that uses translation ids as well for  labels and placeholders

Reply:
I created this tracker to track it: https://tracker.ceph.com/issues/59253